packages/x-date-pickers-pro/src/hooks/useMultiInputRangeField/useMultiInputRangeField.ts

```diff
@@ -0,0 +1,180 @@
+ */
+export function useMultiInputRangeField<
+  TManager extends PickerAnyRangeManager,
+  TForwardedProps extends { [key: string]: any },
+>(parameters: UseMultiInputRangeFieldParameters<TManager, TForwardedProps>) {
+  type TError = PickerManagerError<TManager>;
+
+  const { manager, internalProps, startForwardedProps, endForwardedProps } = parameters;
+
+  const internalPropsWithDefaults = useFieldInternalPropsWithDefaults({
+    manager,
+    internalProps,
+  });
+
+  const {
+    value: valueProp,
+    defaultValue,
+    format,
+    formatDensity,
+    shouldRespectLeadingZeros,
+    onChange,
+    disabled,
+    readOnly,
+    selectedSections,
+    onSelectedSectionsChange,
+    timezone: timezoneProp,
+    enableAccessibleFieldDOMStructure,
+    autoFocus,
+    referenceDate,
+    unstableStartFieldRef,
+    unstableEndFieldRef,
+  } = internalPropsWithDefaults;
+
+  const { value, handleValueChange, timezone } = useControlledValueWithTimezone({
+    name: 'useMultiInputRangeField',
+    timezone: timezoneProp,
+    value: valueProp,
+    defaultValue,
+    referenceDate,
+    onChange,
+    valueManager: manager.internal_valueManager,
+  });
+
+  const { validationError, getValidationErrorForNewValue } = useValidation({
+    props: internalPropsWithDefaults,
+    value,
+    timezone,
+    validator: manager.validator,
+    onError: internalPropsWithDefaults.onError,
+  });
+
+  const buildChangeHandler = (index: 0 | 1): FieldChangeHandler<PickerValue, unknown> => {
+    return (newSingleValue, rawContext) => {
+      const newRange: PickerRangeValue =
+        index === 0 ? [newSingleValue, value[1]] : [value[0], newSingleValue];
+
+      const context: FieldChangeHandlerContext<TError> = {
+        ...rawContext,
+        validationError: getValidationErrorForNewValue(newRange),
+      };
+
+      handleValueChange(newRange, context);
+    };
+  };
+
+  const handleStartDateChange = useEventCallback(buildChangeHandler(0));
+  const handleEndDateChange = useEventCallback(buildChangeHandler(1));
+
+  const selectedSectionsResponse = useMultiInputRangeFieldSelectedSections({
+    selectedSections,
+    onSelectedSectionsChange,
+    unstableStartFieldRef,
+    unstableEndFieldRef,
+  });
+
+  const sharedProps = {
+    disabled,
+    readOnly,
+    timezone,
+    format,
+    formatDensity,
+    shouldRespectLeadingZeros,
+    enableAccessibleFieldDOMStructure,
+  };
+
+  const startDateProps = useMultiInputRangeFieldTextFieldProps<TManager, TForwardedProps>({
+    valueType: manager.valueType,
+    fieldProps: {
+      error: !!validationError[0],
+      ...startForwardedProps,
+      ...selectedSectionsResponse.start,
+      ...sharedProps,
+      value: valueProp === undefined ? undefined : valueProp[0],
+      defaultValue: defaultValue === undefined ? undefined : defaultValue[0],
+      onChange: handleStartDateChange,
+      autoFocus, // Do not add on end field.
```

Comment:
This prop passing looks broken. 🙈 
It end up on the `Stack` element:
![Screenshot 2025-01-14 at 20 28 57](https://github.com/user-attachments/assets/f8b6b11e-9df9-4e1b-984f-bbe83c5859f6)

Reply:
The `autoFocus` was weirdly handled by the field, I moved it to the field internal props to mimic `readOnly` and `disabled`. It seems to work well, even on single input fields but I'll double check :+1:

Reply:
The `autoFocus` behavior seems somewhat broken on v7 as well. 🤔 
But we are now breaking one behavior - `autoFocus` on a `enableAccessibleFieldDOMStructure={false}` SingleInput field. 🙈 
This case works on v7.
As for other cases (`DateRangePicker` or `MultiInputDateRangeField` with `enableAccessibleFieldDOMStructure={false} autoFocus`) - they are also broken on v7, so it's not a blocker and can be fixed separately. 👌

Reply:
Wow standalone field autoFocus is totally broken with the old DOM structure indeed...

But do you have a scenario where it works before this PR and does not work after?
For me standalone single input range field + autoFocus is broken both before and after.

Reply:
> But do you have a scenario where it works before this PR and does not work after?

Nevermind, I must have mixed something up... My previously mentioned case seems to work with both `enableAccessibleFieldDOMStructure` flag versions. 👌